raiden/ui/web/src/app/services/raiden.service.ts

```diff
@@ -169,6 +170,16 @@ export class RaidenService {
         );
     }
 
+    public getPaymentHistory(tokenAddress: string, targetAddress: string): Observable<PaymentEvent[]> {
+        return this.http.get<PaymentEvent[]>(`${this.raidenConfig.api}/payments/${tokenAddress}`)
+            .pipe(
+                flatMap(value => from(value)),
```

Comment:
Please, replace `from+filter+toArray` for simple `Array.filter`, and also `filter` only if `targetAddress` is truty